src/lib/blockchain_snark/blockchain_snark_state.ml

```diff
@@ -224,7 +224,7 @@ let%snarkydef_ step ~(logger : Logger.t)
     in
     let%bind previous_state_hash =
       match constraint_constants.fork with
-      | Some { previous_state_hash = fork_prev; _ } ->
```

Comment:
Did this genuinely need the previous state hash? this can't still be correct with the change to the meaning, right?